runtime/gc_vlhgc/CopyForwardScheme.cpp

```diff
@@ -1606,7 +1608,41 @@ MM_CopyForwardScheme::copyForwardCollectionSet(MM_EnvironmentVLHGC *env)
 	/* Do any final work to regions in order to release them back to the main collector implementation */
 	postProcessRegions(env);
 
-	return copyForwardCompletedSuccessfully(env);
+	static_cast<MM_CycleStateVLHGC*>(env->_cycleState)->_abortFlagRaisedDuringPGC = copyForwardCompletedSuccessfully(env);
+}
+
+#if defined(OMR_GC_VLHGC_CONCURRENT_COPY_FORWARD)
+void
+MM_CopyForwardScheme::concurrentCopyForwardCollectionSet(MM_EnvironmentVLHGC *env)
+{
```

Comment:
You may add an additonal clarifications, that this method is noramlly callled twice for each PGC cycle, once for initial STW increment and once for the final STW increment.
For each of those increments isConcurrentCycleInProgress will change its state/value. For inital increment, it will change from false to true causing only preProcess step being performed, and for the final from true to false causing only postProcess step being performed.

Reply:
That makes sense, I'll update the comments